include/revng/ADT/GenericGraph.h

```diff
@@ -353,11 +370,557 @@ class BidirectionalNode
   }
 
   bool hasPredecessors() const { return Predecessors.size() != 0; }
+  size_t predecessorCount() const { return Predecessors.size(); }
```

Comment:
What do you mean? You want me to also add `neighborCount` (a sum of successors and predecessors) or something?

Note that we need both `predecessorCount` and `successorCount`.

Reply:
> What do you mean? You want me to also add `neighborCount` (a sum of successors and predecessors) or something?

Yes.

Reply:
Now that I'm looking at it more closely, maybe it just makes more sense to remove those `(?:hasN|n)eighbor(?:s|Count)` from the new node rather than adding them to the old one? Considering I don't think I even use those anywhere. Hell, I didn't even remember those existed.

Reply:
I thought you added them for a reason.
I'm OK with dropping them.

Reply:
Well, there are, technically, places where they could be used, but I'm pretty sure I just forgot about this method and used `Node.successorCount() + Node.predecessorCount()` instead. At this point yes, let's just remove it.

Reply:
Done.